homeassistant/components/honeywell/__init__.py

```diff
@@ -1 +1,113 @@
 """Support for Honeywell (US) Total Connect Comfort climate systems."""
+from datetime import timedelta
+
+import somecomfort
+
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.util import Throttle
+
+from .const import _LOGGER, CONF_DEV_ID, CONF_LOC_ID, DOMAIN
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=180)
+PLATFORMS = ["climate"]
+
+
+async def async_setup_entry(hass, config):
+    """Set up the Honeywell thermostat."""
+    username = config.data[CONF_USERNAME]
+    password = config.data[CONF_PASSWORD]
+
+    try:
+        client = await hass.async_add_executor_job(
+            somecomfort.SomeComfort, username, password
+        )
+    except somecomfort.AuthError:
+        _LOGGER.error("Failed to login to honeywell account %s", username)
+        return False
+    except somecomfort.SomeComfortError:
+        _LOGGER.error(
+            "Failed to initialize the Honeywell client: "
+            "Check your configuration (username, password), "
+            "or maybe you have exceeded the API rate limit?"
+        )
+        return False
+
+    loc_id = config.data.get(CONF_LOC_ID)
+    dev_id = config.data.get(CONF_DEV_ID)
+
+    for location in client.locations_by_id.values():
+        for device in location.devices_by_id.values():
+            if (not loc_id or location.locationid == loc_id) and (
+                not dev_id or device.deviceid == dev_id
+            ):
+                data = HoneywellService(hass, client, username, password, device)
+                await data.update()
+                hass.data[DOMAIN] = data
+                hass.config_entries.async_setup_platforms(config, PLATFORMS)
```

Comment:
My understanding is that you can, in theory, have more than 1 thermostat on the RedLINK network (though I only currently have 1 thermostat and 1 outdoor sensor). I think this probably makes sense rather than throwing an error or something if there happens to be more than 1.

Reply:
Upon further reflection, you're correct, this is odd. We're effectively taking a response that may contain multiple locations/devices and attempting to force it into a single location/device. The algorithm as it exists now will only take the last location/device in the array. Ultimately this will still work for me personally since I only have 1 location/device (outdoor sensor values seem to be added as attributes to the indoor thermostat) and it won't cause any run time exceptions. But it should definitely be made more robust. I can do that in a follow PR after this one is merged.